performant-software/digital-fauvel
# Add transcription search

### What does this PR do?
Adds a search feature on top of the Mirador viewer interface, allowing the user to view a list of matches for a search term across all transcriptions and to follow links from those results to their corresponding pages/lines. The implementation uses Lunr for client-side search and includes a script to pre-create a Lunr index from the manifest and annotation-list JSON files. Also includes a minor fix for annotation list parse-ability and moves transcription rendering preferences into Jekyll config settings.

Standing questions for UI: 
- Is the context provided with search results (the line in which the search term occurs and the page slug) adequate?
- Should the search form and results list be moved from the nav area to the Mirador sidebar?

### What issues does it address?
Closes #17, closes #18, closes #19, addresses #20 

### How to test?
- Type a search term into the text field marked "Search"
- Click a result from the list. The Mirador viewer (with a slight delay) should switch to Image View if not already in it, should switch to the page indicated by the search result, and should zoom in to center on the line in which the result appears. The line in which the result appears should have its backdrop turn white.
- Clear the search term by clicking the x icon in the search field. The list of results should disappear, as should the white backdrop in the highlighted line.
- Find a search term that yields results across multiple languages (e.g. "Fauvel"). Clicking a result in a language other than that in which transcriptions are currently rendered on the page should replace the rendered transcriptions with those in the result's language. Try this with results from different pages as well as from the same page.
